src/tyr/isochrone_serializer.cc

```diff
@@ -12,117 +12,193 @@ using namespace valhalla::baldr::json;
 
 namespace {
 using rgba_t = std::tuple<float, float, float>;
+
+using namespace valhalla;
+using namespace tyr;
+using namespace midgard;
+using contour_t = std::list<PointLL>;                 // single ring
+using feature_t = std::list<contour_t>;               // rings per interval
+using contours_t = std::vector<std::list<feature_t>>; // all rings
+using contour_group_t = std::vector<const contour_t*>;
+using grouped_contours_t = std::vector<contour_group_t>;
+using contour_interval_t = std::tuple<size_t, float, std::string, std::string>;
+
+grouped_contours_t GroupContours(const bool polygons, const feature_t& contours) {
+  grouped_contours_t results;
+
+  // if the user requested linestrings, we'll give them linestrings
+  if (!polygons || contours.size() < 2) {
+    std::for_each(contours.begin(), contours.end(),
+                  [&results](const contour_t& c) { results.push_back({&c}); });
```

Comment:
probably we can reserve here with `results.reserve(contours.size()`?

Reply:
They're nested vectors, of which we don't know the size at runtime, so not sure this makes any difference

Reply:
You’re right, didn’t see the nesting